### PR TITLE
bugfix: lint subspec error when the name of subspec start with the po…

### PR DESCRIPTION
**Pull request by XianpuMeng**
fix the bug that lint subspec error when the name of subspec start with the pod name.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* bugfix: lint subspec error when the name of subspec start with the po…  
  [XianpuMeng](https://github.com/XianpuMeng)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* bugfix: lint subspec error when the name of subspec start with the po…  
  [XianpuMeng](https://github.com/XianpuMeng)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -58,6 +58,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Fix lint subspec error when the name of subspec start with the pod name
```
Comment (dnkoutso) [inline CHANGELOG.md:61]: please add two empty spaces at the end of this line for markdown formatting.

File: CHANGELOG.md [hunk 2]
```diff
@@ -58,6 +58,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
...
+  [XianpuMeng](https://github.com/XianpuMeng)
+  [#9914](https://github.com/CocoaPods/CocoaPods/pull/9914)
```
Comment (dnkoutso) [inline CHANGELOG.md:63]: change this to https://github.com/CocoaPods/CocoaPods/issues/9906

**Comment by dnkoutso** [comment]
@XianpuMeng can you please add a test?

**Comment by XianpuMeng** [comment]
> @XianpuMeng can you please add a test?

Pushed! But I cannot find the problem :(
How can I fix it?

**Comment by dnkoutso** [comment]
The subspec you are supplying in the test fixture is not available. You will need to add one that exists in the fixtures directory.

**Comment by XianpuMeng** [comment]
Got it. But there is not an appropriate case in the fixtures directory probably. What can I do? I'm sorry for that I'm a rookie on ruby. :(

**Comment by dnkoutso** [comment]
I will try to take this over.

**Comment by dnkoutso** [comment]
fixed the tests https://github.com/CocoaPods/CocoaPods/pull/10761 and rebased to latest master, closing this. Gave you credit as well!

**Pull request closed without merging**